File: 原創不分類題庫/Contents/e169/e169.cpp
#include<iostream>

using namespace std;

int main(void)
{
	int n, score, total;
	int pick[100] = {0};
	
	while(1)
	{
		cin >> n;
		
		if(!n)
			break;
		
		for(int i = 0; i < n; i++)
		{
			cin >> score;
			pick[score % 100]++;
		}
		
		total = pick[0] * (pick[0] - 1) / 2 + pick[50] * (pick[50] - 1) / 2;
		pick[0] = 0;
		pick[50] = 0;
		
		for(int i = 0; i < 50; i++)
		{
			total += pick[i] * pick[100 - i];
			pick[i] = 0;
			pick[100 - i] = 0;
		}
		
		cout << total << endl;
	}
}

File: 原創不分類題庫/Contents/e169/e169.py
import sys

pick = [0] * 101
for n in sys.stdin:
    n = int(n)

    if n == 0:
        break

    scores = [int(e) for e in sys.stdin.readline().split()]
    for i in scores:
        pick[i % 100] += 1
    
    total = pick[0] * (pick[0] - 1) // 2 + pick[50] * (pick[50] - 1) // 2
    pick[0] = pick[50] = 0
    
    for j in range(50):
        total += pick[j] * pick[100 - j]
        pick[j] = pick[100 - j] = 0

    print(total) 
